### DISCUSSION TITLE
Error: control/dns_config | json Encode: attempted to read more than 65536 bytes | 400

### DISCUSSION THREAD

**Discussion by Gangnamsstyle** [AdguardTeam/AdGuardHome#2814] [Q&A] [answered]
Have a question or an idea? Please search it [on our forum](https://github.com/AdguardTeam/AdGuardHome/discussions) to make sure it was not yet asked. If you cannot find what you had in mind, please [submit it here](https://github.com/AdguardTeam/AdGuardHome/discussions/new).

### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [ ] I am running the latest version YES
- [ ] I checked the documentation and found no answer YES
- [ ] I checked to make sure that this issue has not already been filed YES

### Problem Description

Hi,I have a domain list aourd 110KB,but ADH seems just sopport specify DNS upstream for the specific domain(s) less than 64KB
![image](https://user-images.githubusercontent.com/69400033/111016808-da1f4380-83ea-11eb-828f-94e73db9d2a5.png)

![image](https://user-images.githubusercontent.com/69400033/111016809-da1f4380-83ea-11eb-8145-272ac6a62d9a.png)

Is there any sulution to solve this problems?
Thanks.

**Comment by ameshkov** [comment, marked as answer]
I think you'd better use `upstream_dns_file` setting in `AdGuardHome.yaml` instead:
https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-from-file

**Reply by maidmeow4** [reply]
Can I set multiple `upstream_dns_file`?

**Comment by Gangnamsstyle** [comment]
Thank you very much ,Your sulution solved my problems perfectly.

Andrey Meshkov ***@***.***> 于2021年3月14日周日 下午9:44写道：

**Comment by Eyeborgs** [comment]
@ameshkov но все же в будущем эта ошибка будет исправлена нас счет 65536 bytes? 
потому что эта лишнее время подключатся к VPS чтоб отредактировать файл если это можно легко сделать с Web интерфейса..

**Reply by ameshkov** [reply]
@Eyeborgs а где оно вам надо? Что именно не влезает в 64кб?

**Reply by Eyeborgs** [reply]
@ameshkov я конечно еще не тестировал нас счет Upstream DNS-серверы но раз уже скинули эту ошибку то возможно она есть а значит в будущем я не смогу сделать в Web интерфейсе куча правил DNS и придется делать файл и его подключать в AdGuardHome.yaml а после все время подключатся к хранилищу VPS чтоб изменять файл с правилами Upstream DNS что не удобно..

**Reply by ameshkov** [reply]
Ну многомегабайтные списки апстримов только в Китае нужны. Да и через интерфейс такими вещами очень неудобно управлять - он тормозить будет дико.
